pkg/k8sutil/statefulset.go

```diff
@@ -23,7 +23,9 @@ func WaitForStatefulSetReady(ctx context.Context, clientset kubernetes.Interface
 			return nil
 		}
 
-		if s.Status.ObservedGeneration == s.ObjectMeta.Generation && s.Status.ReadyReplicas == *s.Spec.Replicas {
+		if s.Status.ObservedGeneration == s.ObjectMeta.Generation &&
+			s.Status.ReadyReplicas == *s.Spec.Replicas &&
+			s.Status.UpdateRevision == s.Status.CurrentRevision {
```

Comment:
should we make this change for deployments as well? e.g. the `WaitForDeploymentReady` function.

Reply:
looks like it's only for statefulsets: https://github.com/kubernetes/api/blob/v0.26.1/apps/v1/types.go#L477